quint/src/runtime/impl/compilerImpl.ts

```diff
@@ -1231,18 +1250,37 @@ export class CompilerVisitor implements IRVisitor {
     )
   }
 
-  // The simulator core: produce multiple random runs
-  // and check the given state invariant (state assertion).
-  //
-  // Technically, this is similar to the implementation of folds.
-  // However, it also restores the state and saves a trace, if there is any.
   private test(sourceId: bigint) {
     if (this.compStack.length < 5) {
       this.addCompileError(sourceId,
         'Not enough arguments on stack for "_test"')
       return
     }
 
+    const [nruns, nsteps, init, next, inv] = this.compStack.splice(-5)
+    this.runTestSimulation(nruns, nsteps, init, next, inv)
+  }
+
+  private testOnce(sourceId: bigint) {
+    if (this.compStack.length < 4) {
+      this.addCompileError(sourceId,
+        'Not enough arguments on stack for "_testOnce"')
+      return
+    }
+
+    const [nsteps, init, next, inv] = this.compStack.splice(-4)
+    const nruns = mkConstComputable(rv.mkInt(1n))
+    this.runTestSimulation(nruns, nsteps, init, next, inv)
+  }
+
+  // The simulator core: produce multiple random runs
+  // and check the given state invariant (state assertion).
+  //
+  // Technically, this is similar to the implementation of folds.
+  // However, it also restores the state and saves a trace, if there is any.
+  private runTestSimulation(
```

Comment:
Extracted from `private test` in order to reuse it in `private testOnce`. Mostly left unchanged except that now we don't lookup computables for `init`, `next` and `inv` anymore, as they are computables themselves.